[BUG] ESLint Warnings

### To Reproduce

There are currently ESLint warnings for the project. We should resolve those. I have specified the warnings in the "Additional Context" of this issue.

### Expected Behavior

Ideally we shouldn't have any ESLint warnings.

### Screenshots

_No response_

### Device

Not Applicable

### Browser

Not Applicable

### Operating System

Not Applicable

### Additional Context

```
/home/runner/work/rockylinux.org/rockylinux.org/app/[locale]/download/components/CloudImage/Table/DataTable.tsx
   90:6   warning  React Hook React.useEffect has a missing dependency: 'table'. Either include it or remove the dependency array                                  react-hooks/exhaustive-deps
   90:7   warning  React Hook React.useEffect has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked  react-hooks/exhaustive-deps
  144:18  warning  Strings not allowed in JSX files: "No results."                                                                                                 react/jsx-no-literals
  157:12  warning  Strings not allowed in JSX files: "Previous"                                                                                                    react/jsx-no-literals
  165:12  warning  Strings not allowed in JSX files: "Next"                                                                                                        react/jsx-no-literals

/home/runner/work/rockylinux.org/rockylinux.org/components/ui/dialog.tsx
  49:35  warning  Strings not allowed in JSX files: "Close"  react/jsx-no-literals

✖ 6 problems (0 errors, 6 warnings)```

--- a/app/[locale]/download/components/CloudImage/Table/DataTable.tsx
+++ b/app/[locale]/download/components/CloudImage/Table/DataTable.tsx
@@ -36,10 +36,8 @@ interface DataTableProps<TData, TValue> {
 }
 
 const fuzzyFilter: FilterFn<any> = (row, columnId, value, addMeta) => {
-  // Rank the item
   const itemRank = rankItem(row.getValue(columnId), value);
 
-  // Store the itemRank info
   addMeta({
     itemRank,
   });
@@ -82,12 +80,17 @@ export function DataTable<TData, TValue>({
   });
 
   React.useEffect(() => {
-    if (table.getState().columnFilters[0]?.id === "fullName") {
-      if (table.getState().sorting[0]?.id !== "fullName") {
-        table.setSorting([{ id: "fullName", desc: false }]);
-      }
+    const columnFilters = table.getState().columnFilters[0];
+    const sorting = table.getState().sorting[0];
+
+    if (columnFilters?.id === "fullName" && sorting?.id !== "fullName") {
+      table.setSorting([{ id: "fullName", desc: false }]);
     }
-  }, [table.getState().columnFilters[0]?.id]);
+  }, [table]);
+
+  const noResultsText = "No results.";
+  const previousText = "Previous";
+  const nextText = "Next";
 
   return (
     <div>
@@ -142,7 +145,7 @@ export function DataTable<TData, TValue>({
                   colSpan={columns.length}
                   className="h-24 text-center"
                 >
-                  No results.
+                  {noResultsText}
                 </TableCell>
               </TableRow>
             )}
@@ -155,7 +158,7 @@ export function DataTable<TData, TValue>({
             onClick={() => table.previousPage()}
             disabled={!table.getCanPreviousPage()}
           >
-            Previous
+            {previousText}
           </Button>
           <Button
             variant="outline"
@@ -163,7 +166,7 @@ export function DataTable<TData, TValue>({
             onClick={() => table.nextPage()}
             disabled={!table.getCanNextPage()}
           >
-            Next
+            {nextText}
           </Button>
         </div>
       </div>

--- a/components/ui/dialog.tsx
+++ b/components/ui/dialog.tsx
@@ -29,6 +29,8 @@ const DialogOverlay = React.forwardRef<
 ));
 DialogOverlay.displayName = DialogPrimitive.Overlay.displayName;
 
+const closeButtonText = "Close";
+
 const DialogContent = React.forwardRef<
   React.ElementRef<typeof DialogPrimitive.Content>,
   React.ComponentPropsWithoutRef<typeof DialogPrimitive.Content>
@@ -46,7 +48,7 @@ const DialogContent = React.forwardRef<
       {children}
       <DialogPrimitive.Close className="absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground">
         <Cross2Icon className="h-4 w-4" />
-        <span className="sr-only">Close</span>
+        <span className="sr-only">{closeButtonText}</span>
       </DialogPrimitive.Close>
     </DialogPrimitive.Content>
   </DialogPortal>

